packages/grid_client/tests/modules/applications/casperlabs.test.ts

```diff
@@ -0,0 +1,267 @@
+import axios from "axios";
+import { setTimeout } from "timers/promises";
+
+import { FilterOptions, GatewayNameModel, generateString, GridClient, MachinesModel, randomChoice } from "../../../src";
+import { config, getClient } from "../../client_loader";
+import { bytesToGB, generateInt, getOnlineNode, log, splitIP } from "../../utils";
+
+jest.setTimeout(1500000);
+
+let gridClient: GridClient;
+let deploymentName: string;
+
+beforeAll(async () => {
+  gridClient = await getClient();
+  deploymentName = generateString(15);
+  gridClient.clientOptions.projectName = `casperlabs/${deploymentName}`;
+  gridClient._connect();
+  return gridClient;
+});
+
+//Private IP Regex
+const ipRegex = /(^127\.)|(^10\.)|(^172\.1[6-9]\.)|(^172\.2[0-9]\.)|(^172\.3[0-1]\.)|(^192\.168\.)/;
+
+test("TC2683 - Applications: Deploy Casperlabs", async () => {
```

Comment:
This test does way more than just deploying the instance. I suggest we remove all other code to seperate tests i.e checking for the contract, gateway. Maybe we can add another test for SSHing the deploying. All tests should be done only if the first has passed.

Reply:
The only test made here is to deploy and check the site availability; other assertions are necessary to make sure everything is done correctly and it doesn't take any time in any way.

I will check the possibility of making an SSH test and add in another PR if so.